### DISCUSSION TITLE
Recipient local part in subject

### DISCUSSION THREAD

**Discussion by metrax** [rspamd/rspamd#4224] [Q&A] [answered]
Hi,

i'm receiving a lot of spam with this schema:
![image](https://user-images.githubusercontent.com/1061866/180969217-d6eaa27e-618b-4bb1-b4a5-6e1246bef11e.png)

The local part of the recipent is the first word of the subject.

Is there any simpel way to score this schema higher?

Thank you.

**Comment by moisseev** [comment, marked as answer]
You can do it with [regexp module](https://rspamd.com/doc/modules/regexp.html).

**Reply by metrax** [reply]
Thank for your reply. It was the right hint ;-)
I could realize it with lua:
```
local rspamd_regexp = require "rspamd_regexp"
local util = require "rspamd_util"

rspamd_config:register_symbol{
        name = "LOCAL_PART_SUBJECT",
        type = 'normal',
        callback = function(task)
                local to = task:get_header('To')
                local sbj = task:get_header('Subject')
                local lp = util.parse_mail_address(to, task:get_mempool())
                local subject_re = rspamd_regexp.create_cached("/^"..lp[1]["user"].."/i")
                local stripped_subject =  subject_re:search(sbj, false, true)
                if stripped_subject then
                        return true
                else
                        return false
                end
        end,
        score = 3.0,
        group = 'subject',
        description = 'Subject starts with local part of To',

}
```

**Reply by vstakhov** [reply]
I would suggest at least to replace `rspamd_regexp.create` with `rspamd_regexp.create_cached` due to performance considerations...

**Reply by metrax** [reply]
@vstakhov thank you very much fror your suggestion! I overtook it to my instance and edited the post here for other users.

**Comment by Bierchermuesli** [comment]
nice one, thanks for sharing!
for the others: put this in `$CONFDIR/rspamd.local.lua ` (`/etc/rspamd/rspamd.local.lua`)

I think this rule is very effective, would't it be good for upstream?

Here a little modification to match anywhere in the subject, like `Subject:  Re: mmuster, Paket??`: 


```
local rspamd_regexp = require "rspamd_regexp"
local util = require "rspamd_util"

rspamd_config:register_symbol{
        name = "RCPT_IN_SUBJECT",
        type = 'normal',
        callback = function(task)
                local to = task:get_header('To')
                local sbj = task:get_header('Subject')
                local lp = util.parse_mail_address(to, task:get_mempool())
                local subject_re = rspamd_regexp.create_cached("/^"..lp[1]["user"].."/i")
                local stripped_subject =  subject_re:search(sbj, false, true)
                if stripped_subject then
                        return true
                else
                        return false
                end
        end,
        score = 3.0,
        group = 'subject',
        description = 'Subject contains recipients local part in To',
}
```

**Reply by metrax** [reply]
@Bierchermuesli 
This version of my LUA Rule is ending into a OOM Kill, if the To-Header is undiscloded.
Please have a look here #4437 for a better version.

**Comment by abonadt** [comment]
Thank you for sharing. We have another approach to handle those E-Mails. I found out that spammers put randomly special characters in words. So a simple regexp filter out most of those spams: [a-zA-Z][\\#\\$\\%\\^\\*][a-hj-qs-zA-HJ-UW-Z]
I had to exclude some letters to allow german gender forms like "*innen" and others. Comments are welcome - we are still testing.

**Comment by kpapad904** [comment]
Hi, could this approach (lua + regexp) be used to catch phishing mails, where the From: field (in either the EmailAddress or the DisplayName parts) contains certain variations of "brand" name, but the Return-Path: (SpamAssassin uses the term EnvelopeFrom:) isn't also from brand.tld ?

In recent years, 90% of the bank phishing mails in my Inbox are of this type:
```
From brand-comunication @t-online.de  Mon Sep 27 14:12:01 2021
From brand.mail-contact @magenta.de  Fri Nov  5 00:49:05 2021

From: "BRAND.myBank" <support @sharpshooter.rgr.jp>
From: "BRAND.myBank" <support @toshima-theatre.jp>
From: myBankBRAND <support @sharpshooter.rgr.jp>
From: BRAND <support @hayashi-dentalclinic.jp>
```
About 1.5yr ago I wanted to move from SA to Rspamd but couldn't migrate these local rules.

Please check discussion at https://github.com/rspamd/rspamd/discussions/4042
